storybookjs/storybook
# Docs: Update useSWC API reference

## What I did

Related to #22861, this PR updates the docs to be more API reference-centric.

- Add to API reference
- Update related content/snippets

## How to test

1. Follow the steps in the [contributing instructions](https://storybook.js.org/docs/react/contribute/new-snippets#preview-your-work) for this branch, `api-reference-use-swc`
2. Check for:
    a. Completeness
        - Is every property documented fully?
    b. Correctness
        - Types
        - Required or not
        - Default values
        - Descriptions
        - Example snippets
    c. Consistency in structure

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
